lib/common/xml.c

```diff
@@ -686,12 +698,26 @@ create_xml_node(xmlNode * parent, const char *name)
 
     if (parent == NULL) {
         doc = xmlNewDoc((pcmkXmlStr) "1.0");
+        if (doc == NULL) {
+            return NULL;
```

Comment:
We don't usually check create_xml_node() for NULL either, but that can be left for blue skies.

Presumably if we return NULL past this point, we have some freeing to do

I'm tempted to just assert on memory errors in create_xml_node(), even though I usually prefer returning an indication when possible

Reply:
I'm good either way. In the past my default approach was to assert on memory errors, like GLib does. You've said recently that you're moving away from doing that, since a core dump isn't useful.

I'd been meaning to ask you how your feelings translated to functions like `pcmk__str_update()`.

Reply:
My general feeling is that library functions in particular should bubble up errors since callers have more context to know the best reaction. But something like pcmk__str_update() is so widely used that checking the return value all the time would be a major hassle, both in initially converting it and in readability. So I don't like it but I accept it :)

Reply:
I'm leaving this alone for now; I'm not sure where we landed on changing it. Could do either:
* leave it alone for now, maybe make a wishlist task for NULL-checking `create_xml_node()`, or
* assert on memory error

Reply:
It's fine as is, except that we need to free any allocated memory before returning NULL